mesheryctl/internal/cli/root/app/onboard_test.go

```diff
@@ -0,0 +1,121 @@
+package app
+
+import (
+	"bytes"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+)
+
+func TestOnboardCmd(t *testing.T) {
+	// setup current context
+	utils.SetupContextEnv(t)
+
+	// initialize mock server for handling requests
+	utils.StartMockery(t)
+
+	// create a test helper
+	testContext := utils.NewTestHelper(t)
+
+	// get current directory
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("Not able to get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	fixturesDir := filepath.Join(currDir, "fixtures")
+
+	// test scenrios for fetching data
+	tests := []struct {
+		Name             string
+		Args             []string
+		ExpectedResponse string
+		URLs             []utils.MockURL
+		Token            string
+		ExpectError      bool
+	}{
+		{
```

Comment:
can we add one more test-case with `--skip-save` flag?

Reply:
@piyushsingariya  I have added test for --skip-save as well